config-controller/pkg/client/client_test.go

```diff
@@ -128,6 +128,48 @@ func TestCachedClientGet(t *testing.T) {
 	assert.False(t, exists, "Policy exists when it should not")
 }
 
+// TestCachedClientDelete validates that the cached client deletes policies as expected
+func TestCachedClientDelete(t *testing.T) {
+	newPolicy := storage.Policy{
+		Name:            "This is a new policy",
+		Description:     "A really good description",
+		LifecycleStages: []storage.LifecycleStage{storage.LifecycleStage_BUILD},
+		Severity:        storage.Severity_CRITICAL_SEVERITY,
+		PolicySections: []*storage.PolicySection{{
+			SectionName: "Section A",
+			PolicyGroups: []*storage.PolicyGroup{{
+				FieldName: "Image",
+				Values: []*storage.PolicyValue{{
+					Value: "hello",
+				}},
+			}},
+		}},
+	}
+
+	mockPolicyToReturn := newPolicy.CloneVT()
+	mockPolicyToReturn.Id = "abc123"
+	mockPolicyToReturn.Source = storage.PolicySource_DECLARATIVE
+
+	clientTest := setUp(t, func(mockClient *mocks.MockPolicyClient, policies []*storage.Policy) {
+		mockClient.EXPECT().ListPolicies(gomock.Any()).Return(createListPolicies(policies), nil).Times(1)
+		mockClient.EXPECT().GetPolicy(gomock.Any(), policies[0].Id).Return(policies[0], nil).Times(1)
+		mockClient.EXPECT().GetPolicy(gomock.Any(), policies[1].Id).Return(policies[1], nil).Times(1)
+		mockClient.EXPECT().PostPolicy(gomock.Any(), &newPolicy).Return(mockPolicyToReturn, nil).Times(1)
+		mockClient.EXPECT().DeletePolicy(gomock.Any(), mockPolicyToReturn.Id).Return(nil).Times(1)
+		mockClient.EXPECT().TokenExchange(gomock.Any()).Return(nil).Times(1)
+	})
+	defer clientTest.controller.Finish()
+
+	createdPolicy, err := clientTest.client.CreatePolicy(clientTest.ctx, &newPolicy)
+
+	assert.NoError(t, err, "Unexpected error creating a policy")
+	assert.Equal(t, "abc123", createdPolicy.Id)
+
+	err = clientTest.client.DeletePolicy(clientTest.ctx, newPolicy.Name)
+	assert.NoError(t, err, "Unexpected error deleting the policy")
```

Comment:
would it be reasonable to list the policies after this line and ensure the deleted policy ID isn't in the list?

Reply:
no, since everything programmed in the mocks. If we had a way to look into the cache, that would be useful but we don't.